propagators/opentelemetry-propagator-aws-xray/src/AWSXRayPropagator.ts

```diff
@@ -0,0 +1,200 @@
+    if (!isValid(spanContext)) return context;
+
+    return setExtractedSpanContext(context, spanContext);
+  }
+  
+  private getSpanContextFromHeader(carrier: unknown, getter: GetterFunction): SpanContext {
+    const traceHeader = getter(carrier, AWSXRAY_TRACE_ID_HEADER);
+    // Only if the returned traceHeader is no empty string can be extracted
+    if (!traceHeader || typeof traceHeader !== 'string') return INVALID_SPAN_CONTEXT;
+
+    let pos = 0, trimmedPart: string;
+    let parsedTraceId = INVALID_TRACE_ID;
+    let parsedSpanId = INVALID_SPAN_ID;
+    let parsedTraceFlags = null;
+    while(pos < traceHeader.length) {
+      const delimiterIndex = traceHeader.indexOf(TRACE_HEADER_DELIMITER, pos);
+      if (delimiterIndex >= 0) {
+        trimmedPart = traceHeader.substring(pos, delimiterIndex).trim();
+        pos = delimiterIndex + 1;
+      } else {
+        //last part
+        trimmedPart = traceHeader.substring(pos).trim();
+        pos = traceHeader.length;
+      }
+      const equalsIndex = trimmedPart.indexOf(KV_DELIMITER);
+
+      const value = trimmedPart.substring(equalsIndex + 1);
+
+      if (trimmedPart.startsWith(TRACE_ID_KEY)) {
+        parsedTraceId = this._parseTraceId(value);
+      }
+      if (trimmedPart.startsWith(PARENT_ID_KEY)) {
+        parsedSpanId = this._parseSpanId(value);
+      }
+      if (trimmedPart.startsWith(SAMPLED_FLAG_KEY)) {
+        parsedTraceFlags = this._parseTraceFlag(value);
+      }
+    }
+    if (parsedTraceFlags === null) {
+      return INVALID_SPAN_CONTEXT;
+    }
+    const resultSpanContext: SpanContext = {
+      traceId: parsedTraceId,
+      spanId: parsedSpanId,
+      traceFlags: parsedTraceFlags,
+      isRemote: true,
+    }
+    if (!isValid(resultSpanContext)) {
+      return INVALID_SPAN_CONTEXT;
+    }
+    return resultSpanContext;
+  }
+
+  private _parseTraceId(xrayTraceId: string): string {
+    // Check length of trace id
+    if (xrayTraceId.length !== TRACE_ID_LENGTH) {
+      return INVALID_TRACE_ID;
+    }
+
+    // Check version trace id version
+    if (!xrayTraceId.startsWith(TRACE_ID_VERSION)) {
+      return INVALID_TRACE_ID;
+    }
+
+    // Check delimiters
+    if (xrayTraceId.charAt(TRACE_ID_DELIMITER_INDEX_1) !== TRACE_ID_DELIMITER
+    || xrayTraceId.charAt(TRACE_ID_DELIMITER_INDEX_2) !== TRACE_ID_DELIMITER) {
+      return INVALID_TRACE_ID;
+    }
+
+    const epochPart = xrayTraceId.substring(TRACE_ID_DELIMITER_INDEX_1 + 1, TRACE_ID_DELIMITER_INDEX_2);
+    const uniquePart = xrayTraceId.substring(TRACE_ID_DELIMITER_INDEX_2 + 1, TRACE_ID_LENGTH);
+    const resTraceId = epochPart + uniquePart;
+
+    // Check the content of trace id
+    if (!VALID_TRACEID_REGEX.test(resTraceId)) {
+      return INVALID_TRACE_ID
+    }
+
+    return resTraceId;
+  }
+
+  private _parseSpanId(xrayParentId: string): string{
+    return (VALID_SPANID_REGEX.test(xrayParentId))? xrayParentId: INVALID_SPAN_ID;
```

Comment:
style nit: we typically use spaces surrounding the `?` and `:`
```suggestion
    return VALID_SPANID_REGEX.test(xrayParentId) ? xrayParentId : INVALID_SPAN_ID;
```

Reply:
Yes, will do it! Really thank you for your advice! 
Another thing about lint I am pretty confused is that, I followed file structure of jaegerHttpTracePropagator, but I cannot pass the ```npm run lint``` test of GA. I checked eslint file and found it seems to run ```eslint . --ext .ts``` by using eslint.config.js file. eslint.config.js file does exist but still fails, I am blocked here and have no idea what to do.

Reply:
oh my fault, it seems ```npm run test``` command cannot be run locally but GA does this successfully and gives me error info, I will fix lint errors and make another commit.